robot-name/robot-name_test.spec.js

```diff
@@ -12,9 +12,17 @@ describe("Robot", function() {
   });
 
   it("different robots have different names", function() {
-    var robotOne = new Robot();
-    var robotTwo = new Robot();
-    expect(robotOne.name).not.toEqual(robotTwo.name);
+    var usedNames= {};
+
+    var i = 0,
+        max = 10000;
+
+    for (i; i < max; i++) {
```

Comment:
In other languages is common practice to declare a variable as closest as its use as possible. In JavaScript it is recommended to declare them at the beginning of the scope (usually the enclosing `function`, for loops do not create a new scope, as other languages do). JSLint just try to force this rule.

In fact, the recomendation is to declare all variables inside a the same scope in just one `var` statement, see [Move `var` declarations to the top of the function][1].

The reason for this rule is the way JavaScript works with variables (see [*hoisting*][2]). At execution time, JavaScript gets all variables and creates them at the top of the scope. Then, JavaScript assign values where the programmer initializes the variable. This can cause some problems:

```javascript
function output() {
  console.log(foo);
  var foo = 'bar';
  console.log(foo);
}
output();
```

Other languages would throw an error on the first `console.log(foo)`, but the output of this *program* is:

```bash
undefined
bar
```

The way JavaScript works, the previous code is equivalent to this:

```javascript
function output() {
  var foo;
  console.log(foo);
  foo = 'bar';
  console.log(foo);
}
output();
```

All this is just a recommendation, to prevent silly mistakes. So, it would be fine to write the `for` loop in this way:

```javascript
for(var i = 0; i < max; i++) { /* ... */ }
```

as long as you know it won't cause any problem.

[1]: http://jslinterrors.com/move-var-declarations-to-the-top-of-the-function/
[2]: http://www.w3schools.com/js/js_hoisting.asp